tasks.py

```diff
@@ -70,30 +71,23 @@ def _validate_python_version(line):
 
 @task
 def install_minimum(c):
-    with open('pyproject.toml', 'r') as pyproject:
-        lines = pyproject.read().splitlines()
+    with open('pyproject.toml', 'r', encoding='utf-8') as pyproject_file:
+        pyproject_data = toml.load(pyproject_file)
 
+    dependencies = pyproject_data.get('project', {}).get('dependencies', [])
     versions = []
-    started = False
-    for line in lines:
-        if started:
-            if line == ']':
-                started = False
-                continue
-
-            line = line.strip()
-            if _validate_python_version(line):
-                requirement = re.match(r'[^>]*', line).group(0)
-                requirement = re.sub(r"""['",]""", '', requirement)
-                version = re.search(r'>=?(\d\.?)+\w*', line).group(0)
-                if version:
-                    version = re.sub(r'>=?', '==', version)
-                    version = re.sub(r"""['",]""", '', version)
-                    requirement += version
-                versions.append(requirement)
-
-        elif (line.startswith('dependencies = [')):
-            started = True
+    for line in dependencies:
+        line = line.strip()
+        if _validate_python_version(line):
+            requirement = re.match(r'[^>]*', line).group(0)
+            requirement = re.sub(r"""['",]""", '', requirement)
+            version = re.search(r'>=?(\d\.?)+\w*', line).group(0)
+            if version:
+                version = re.sub(r'>=?', '==', version)
+                version = re.sub(r"""['",]""", '', version)
+                requirement += version
+
+            versions.append(requirement)
```

Comment:
It might be better to use the Requirement and Specifier class from the packaging library, rather than using regex. The code might be something like this:
```suggestion
    for line in dependencies:
        min_package = find_min_requirement(line)
        min_requirements.append(str(min_package))
    min_requirements = "\n".join(min_requirements) + "\n"
    
from packaging.requirements import Requirement
from packaging.specifiers import Specifier

def find_min_requirement(requirement):
    if is_requirement_path(requirement):
        # skip requirement paths
        # ex '-r core_requirements.txt'
        return
    requirement = remove_comment(requirement)
    if not verify_python_environment(requirement):
        return
    if ">=" in requirement:
        # minimum version specified (ex - 'package >= 0.0.4')
        package = Requirement(requirement)
        version = find_operator_version(package, ">=")
        minimum = create_strict_min(version)
    elif "==" in requirement:
        # version strictly specified
        package = Requirement(requirement)
        version = find_operator_version(package, "==")
        minimum = create_strict_min(version)
    else:
        # minimum version not specified (ex - 'package < 0.0.4')
        # version not specified (ex - 'package')
        raise ValueError(
            "Operator does not exist or is an invalid operator (such as '<'). Please specify the minimum version ('>=', '==')."
        )
    name = determine_package_name(package)
    min_requirement = Requirement(name + str(minimum))
    return min_requirement
      
def is_requirement_path(requirement):
    if ".txt" in requirement and "-r" in requirement:
        return True
    return False

def remove_comment(requirement):
    if "#" in requirement:
        # remove everything after comment character
        requirement = requirement.split("#")[0]
    return requirement
    
def determine_package_name(package):
    name = package.name
    if len(package.extras) > 0:
        name = package.name + "[" + package.extras.pop() + "]"
    return name
    
def verify_python_environment(requirement):
    package = Requirement(requirement)
    if not package.marker:
        # no python version specified in requirement
        return True
    elif package.marker and package.marker.evaluate():
        # evaluate --> evaluating the given marker against the current Python process environment
        return True
    return False
```

Reply:
I think I agree with trying to remove regex if possible. One thing I'm wondering that doesn't really come into play in SDGym but might in other libraries is that sometimes we point to a branch directly to install from. If so, would the minimum tests crash there because that syntax doesn't use '<'? An example would be something like
```
dependencies = [
  "<packagename>@git+<url-to-repo>#egg=<branch or hash>",
]
```